dask/bag/core.py

```diff
@@ -183,6 +183,42 @@ def finalize_item(results):
     return results[0]
 
 
+def merge_kwargs(dsk, kwargs):
+    """ Merges dask values from kwargs with dsk.
+
+    Returns the merged dsk and a graph expression useful for use with apply in
+    a dask graph to pass the final computed **kwargs to a function.
+
+    Currently only dask.bag.Item and python literal values are supported.
+    """
+    kw_pairs = []
+    for key, val in iteritems(kwargs):
+        if isinstance(val, Item):
+            dsk = merge(dsk, val.dask)
+            val = val.key
+        # TODO elif isinstance(val, Value):
+        elif isinstance(val, Base):
+            raise NotImplementedError(
+                '%s not supported as kwarg value to Bag.map_partitions'
+                % type(val).__name__)
+        kw_pairs.append([key, val])
+    return dsk, kw_pairs
```

Comment:
OK, makes sense.  Can I put you on the hook to handle this if a user ever asks about why this doesn't work with `Value` objects?

Regarding `dict` it looks like that was a mistake on my part.  I was assuming that the dask scheduler walked down dict objects.  Looks like it doesn't at the moment.

```python
In [1]: import dask

In [2]: def inc(x):
    return x + 1
   ...: 

In [3]: dsk = {'x': 1, 'y': {'a': 'x'}}

In [4]: dask.get(dsk, 'y')
Out[4]: {'a': 'x'}
```